src/align/LinearAlignment.cpp

```diff
@@ -91,10 +91,13 @@ const AlignConfig config(params, AlignMode::GLOBAL);
 std::string NWTranscript(const std::string& target, int j1, int j2, const std::string& query,
                          int i1, int i2, int* score)
 {
-    assert((i1 <= i2) && (j1 <= j2));
+    // This assert does not hold. If j1 > j2 or i1 > i2, means the respective 
+    // subtarget or subquery is empty, ergo we have pure insertions or deletions.
+    // assert((i1 <= i2) && (j1 <= j2)); 
+    
```

Comment:
Is the empty interval always represented in this algorithm by (x, x-1)?  I think it is, though I haven't proven it to myself yet.   If so, we can restore an assert on ``(((i2 >= i1 - 1) && (j2 >= j2 - 1))`` ... and we also would not need a special case.